api/pkg/service/resource/resource.go

```diff
@@ -143,6 +138,27 @@ func (s *service) ByTypeName(ctx context.Context, p *resource.ByTypeNamePayload)
 	return s.resourcesForQuery(q)
 }
 
+// Finds a resource information using it's id
+func (s *service) ByID(ctx context.Context, p *resource.ByIDPayload) (res *resource.Resource, err error) {
+
+	q := s.db.Scopes(withResourceDetails,
+		findByID(p.ID))
+
+	var one model.Resource
+	if err := findOne(q, &one); err != nil {
+		return nil, err
+	}
+
+	res = &resource.Resource{}
+	res = initResource(one)
+
+	for _, v := range one.Versions {
+		res.Versions = append(res.Versions, minVersionInfo(v))
+	}
+
+	return res, nil
+}
+
 func findOne(db *gorm.DB, result interface{}) error {
```

Comment:
Could you please move `findOne` and `findByID` -> `filterByID` and keep between `orderByVersion` and `filterByType`?

Reply:
so that all gorm related func could be together